File: modules/swagger-servlet/src/main/scala/com/wordnik/swagger/servlet/config/ServletScanner.scala
package com.wordnik.swagger.servlet.config

import com.wordnik.swagger.annotations.Api
import com.wordnik.swagger.core.SwaggerContext
import com.wordnik.swagger.config._
import com.wordnik.swagger.reader._

import org.reflections.Reflections
import org.reflections.scanners.{ SubTypesScanner, TypeAnnotationsScanner }
import org.reflections.util.{ ClasspathHelper, ConfigurationBuilder}

import scala.collection.JavaConverters._

class ServletScanner extends Scanner {
  private var resourcePackage: String = _

  def getResourcePackage():String = this.resourcePackage

  def setResourcePackage(resourcePackage: String) = {
    this.resourcePackage = resourcePackage
  }

  def classes(): List[Class[_]] = {
    val config = new ConfigurationBuilder().setUrls(ClasspathHelper.forPackage(resourcePackage)).setScanners(
      new TypeAnnotationsScanner(), new SubTypesScanner())
    new Reflections(config).getTypesAnnotatedWith(classOf[Api]).asScala.toList
  }
}

File: modules/swagger-servlet/src/main/java/io/swagger/servlet/config/ServletScanner.java
package io.swagger.servlet.config;

import io.swagger.annotations.Api;
import io.swagger.config.Scanner;

import org.reflections.Reflections;

import java.util.Set;

import javax.servlet.ServletConfig;

public class ServletScanner implements Scanner {

    private boolean prettyPrint = false;
    private String resourcePackage;

    public ServletScanner(ServletConfig servletConfig) {
        resourcePackage = servletConfig.getInitParameter("swagger.resource.package");
        final String shouldPrettyPrint = servletConfig.getInitParameter("swagger.pretty.print");
        if (shouldPrettyPrint != null) {
            prettyPrint = Boolean.parseBoolean(shouldPrettyPrint);
        }
    }

    public String getResourcePackage() {
        return resourcePackage;
    }

    public void setResourcePackage(String resourcePackage) {
        this.resourcePackage = resourcePackage;
    }

    @Override
    public Set<Class<?>> classes() {
        return new Reflections(resourcePackage).getTypesAnnotatedWith(Api.class);
    }

    @Override
    public boolean getPrettyPrint() {
        return prettyPrint;
    }

    @Override
    public void setPrettyPrint(boolean shouldPrettyPrint) {
        this.prettyPrint = shouldPrettyPrint;
    }
}
